### PR TITLE
Add record_ixd to store_block message

### PR DESCRIPTION
**Pull request by DG12**
Add Prerequisites to README

Change wording from eraseing to freed and block to record and add LOG for new record . Add record_ixd
```
if (RD_SUCCESS == err_code)
        {
             char msg[128];
             snprintf (msg, sizeof (msg), "store_block:freed old block #%d\r\n", record_idx);
             LOG(msg);
        }
        else
        {
             char msg[128];
             snprintf (msg, sizeof (msg), "store_block:creating new block #%d\r\n", record_idx);
             LOG(msg);
        }
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,10 +72,10 @@ The report can be found under _build/artifacts/gcov_.
 Travis will fail the build if unit test fails and Gcov results will get pushed to SonarCloud.
 
 # Usage
-Compile and flash the project to your board using SES if you're developing. 
+Compile and flash the project to your board using Segger Embedded Studio if you're developing. 
 Note: You should erase your board entirely in case there is a bootloader from a previous firmware.
 
-If you're building binaries for distribution, you can and should use the make script to generate binaries.
+If you're building binaries for distribution, you can and should use THE make script to generate binaries.
```
Comment (ojousima) [inline README.md:78]: `THE` isn't proper English

File: ReadDG12.md [hunk 2]
```diff
@@ -0,0 +1 @@
+ReadDG12.md
```
Comment (ojousima) [inline ReadDG12.md:1]: Is this file intentionally included?

**Review by ojousima — CHANGES_REQUESTED** [review]

**Comment by ojousima** [comment]
ok to test

File: ReadDG12.md (hunk 2 above)
Comment (DG12) [reply ReadDG12.md:1]: I intended this as a place holder unless you prefer it not be included or can you suggest another name for a secondary readme

File: README.md (hunk 1 above)
Comment (DG12) [reply README.md:78]: OK how about "the provided script"

Comment (ojousima) [reply README.md]: Looks good

**Review by ojousima — CHANGES_REQUESTED** [review]
Thanks, please remove the `ruuvi_open_private.pem`. It's fetched from releases by build script, having private keys in repository triggers unnecessary false alarms from security tools.

**Comment by ojousima** [comment]
> @otso I guess this is what you are talking about

Yes, exactly

**Comment by ojousima** [comment]
Just checking, for what the Xcode you added to readme is needed? I have it installed on my PC so I might have missed that it's required for something in this project.

p.s. Did you notice https://github.com/DG12/ruuvi.firmware.c/pull/1 ?

File: README.md [hunk 3]
```diff
@@ -14,24 +17,50 @@ Ruuvi Firmware version 3. Built on top of Nordic SDK 15, uses both Ruuvi and ext
 Under development, please follow [Ruuvi Blog](https://blog.ruuvi.com) for details. The project is in beta stage, no breaking changes are intented but will be done if absolutely necessary for some reason.  
 
 # Setting up
+
+## Prerequisites
+* gcc
+* [gcc-arm-none-eabi-7-2018-q2-update](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads/7-2018-q2-update) CHeck that nRF5_SDK_15.3.0_59ac345/components/toolchain/gcc  Makefile.posix or Makefile.windows points to it.
```
Comment (ojousima) [inline README.md:23]: CHeck -> Check

File: README.md [hunk 4]
```diff
@@ -14,24 +17,50 @@ Ruuvi Firmware version 3. Built on top of Nordic SDK 15, uses both Ruuvi and ext
...
+
+### Suggested 
+
+* [Ruuvi Dev kit board](https://shop.ruuvi.com/product/devkit/) and a USB power & data cable.
+* [Nordic nrfjprog](https://www.nordicsemi.com/Software-and-tools/Development-Tools/nRF-Command-Line-Tools) Erase, write and read from Nordic nRF5 System On Chip
+ * [Segger j-link](https://www.segger.com/downloads/jlink) package incluing j-link and JLinkRTTClient
+ 
+### Optional
+* [Ceedling](http://www.throwtheswitch.org/ceedling) Needs Ruby and gcc
+* [PVS-Studio Analyzer](https://www.viva64.com/en/pvs-studio/) Purchase
```
Comment (ojousima) [inline README.md:41]: PVS-Studio gives free licenses for open source projects hosted on GitHub, GitLab or BitBucket: https://www.viva64.com/en/b/0600/

**Review by ojousima — CHANGES_REQUESTED** [review]
I can make the readme fixes too, let me know if you want me to take care of it

File: README.md (hunk 4 above)
Comment (DG12) [reply README.md:41]: Please do make the changes. I just got tired of fighting with GitHub formatting :-(

Comment (ojousima) [reply README.md:41]: ok

**Pull request merged**